src/ARAnchorManager.js

```diff
@@ -0,0 +1,114 @@
+    };
+    // Do not allow to create an anchor for an object3d that already has one.
+    if (this.object3DsToAnchors_.has(object3d)) {
+      return this;
+    }
+    // If the object has a parent and its parent has a parent, it is very
+    // likely that it may not be in world space so warn the developer.
+    // NOTE: Commented out for now because if we use anchors for planes in
+    // ARPlanes, they are added to a identity main parent object but does not
+    // mean they are not in world space. Getting this warning at all times is
+    // not optimal.
+//    if (object3d.parent && object3d.parent.parent) {
+//      console.warn("anchors should be added for Object3D-s in world space");
+//    }
+    // Calculate the model matrix of the object3d using the position and
+    // quaternion of the object.
+    this.scale_.set(1, 1, 1);
+    this.matrix_.compose(object3d.position, object3d.quaternion, this.scale_);
+    const anchor = this.vrDisplay_.addAnchor(this.matrix_.elements);
+    // Store the anchor and the object3d in the structure to retrieve them
+    // later on.
+    this.anchorsToObject3Ds_.set(anchor, object3d);
+    this.object3DsToAnchors_.set(object3d, anchor);
+    return this;
+  }
+
+  /**
+   * Delete anchor associated to the THREE.Object3D.
+   */
+  remove(object3d) {
+    if (!(object3d instanceof Object3D)) {
+      throw new Error('Invalid Object3D trying to remove anchor');
+    };
+    const anchor = this.object3DsToAnchors_.get(object3d);
+    if (!anchor) { 
+      return this;
+    };
+    this.anchorsToObject3Ds_.delete(anchor);
+    this.object3DsToAnchors_.delete(object3d);
+    this.vrDisplay_.removeAnchor(anchor);
+    return this;;
+  }
+
+  /**
+   * Called when any anchors are updated. The updated modelMatrix is
+   * set for the THREE.Object3D associated with each updated anchor.
+   * An 'object3dsupdated' event is emitted with the updated THREE.Object3Ds.
+   */
+  onAnchorsUpdated_(event) {
+    const updatedObject3Ds = [];
+    for (let anchor of event.anchors) {
+      const object3d = this.anchorsToObject3Ds_.get(anchor);
+      if (!object3d) { 
+        // It could happen that there are more than one AnchorManager and
+        // that create Anchors that reside in the native side and are not
+        // handled by all of the AnchorManagers, so let's just continue to
+        // see if the updated anchors are managed by this AnchorManager.
+        continue;
+      };
+      // get the anchor's updated model matrix
+      object3d.matrix.fromArray(anchor.modelMatrix);
+      // decompose the updated matrix into position, orientation, and scale
+      object3d.matrix.decompose(object3d.position,
+                                object3d.quaternion,
+                                this.scale_);
+      updatedObject3Ds.push(object3d);
+    }
+    this.dispatchEvent({
+        type: 'anchorsupdated', 
+        object3ds: updatedObject3Ds});
```

Comment:
I still think `object3ds` for a property name is rather strange; could this just be `objects`?

Reply:
Said and done. It will be fixed in the next push.